tools/compose-image-yml.sh

```diff
@@ -3,7 +3,13 @@
 set -e
 
 yq() {
-  docker run -i --rm -v "${PWD}/":/workdir -w /workdir mikefarah/yq:4.40.5 "$@"
+  if yq --version | grep -cq v4.4
+  then
+    yq "$@"
+    return
+  fi
+
+  go run github.com/mikefarah/yq/v4@v4.40.5 "$@"
```

Comment:
This works for pipeline runs; will it work as well for desktop users? Is it fair to assume that they have go?

Maybe we can capture all cases:

```sh
  if yq --version | grep -cq v4.4
  then
    yq "$@"
    return
  elif go --version 2>/dev/null
  then
    go run github.com/mikefarah/yq/v4@v4.40.5 "$@"
  else
    docker run -i --rm -v "${PWD}/":/workdir -w /workdir mikefarah/yq:4.40.5 "$@"
  fi
```

I might have gotten the syntax a bit wrong. The idea is that we start with your system yq if version is enough, fall back to `go run` if you have it, fall back to docker. Desktop users (real humans) will use whatever they have, while the pipeline always will use the go one (since you ensured it has it).

What do you think?

Reply:
I think it is fair to assume that they have go (because of `get-deps` and `compare-sbom-sources`).

Nevertheless I think it does not hurt to do it this way.

Reply:
I know, let's write a custom bit of software for it, in an unknown language, and...  OK, maybe not. 😁